openraft/src/raft.rs

```diff
@@ -116,7 +116,7 @@ struct RaftInner<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>>
     marker_s: std::marker::PhantomData<S>,
 
     /// The error that cause RaftCore to quit.
-    core_error: std::sync::Mutex<Option<Fatal<C::NodeId>>>,
+    core_error: std::sync::Mutex<Result<(), Fatal<C::NodeId>>>,
```

Comment:
`core_error` needs to be renamed to reflect its purpose.

Reply:
like 'unrecover_error'?

Reply:
core_result? it is a result

Reply:
Done!